test/interleaver/do_interleave_test.exs

```diff
@@ -0,0 +1,67 @@
+defmodule Membrane.DdInterleaveTest do
+  @moduledoc """
+  Tests for DoInterleave module.
+  """
+
+  use ExUnit.Case, async: true
+
+  alias Membrane.AudioMixer.DoInterleave
+
+  require Membrane.Logger
+
+  describe "DoInterleaver interleave should" do
+    defp to_pad(key, queue, end_of_stream \\ false) do
+      {{Membrane.Pad, :input, key}, %{end_of_stream: false, queue: queue}}
+    end
+
+    test "correctly interleave and update queues" do
+      pads =
+        Map.new([
+          to_pad(1, <<1, 2, 3, 4, 5, 6, 7, 8>>),
+          to_pad(2, <<90, 100, 110, 120, 130, 140, 150, 160>>)
+        ])
+
+      order = [1, 2]
+      bytes_per_channel = 4
+
+      interleaved1 = <<1, 90, 2, 100, 3, 110, 4, 120>>
+      interleaved2 = <<1, 2, 90, 100, 3, 4, 110, 120>>
+      interleaved3 = <<1, 2, 3, 90, 100, 110, 4, 120>>
+      interleaved4 = <<1, 2, 3, 4, 90, 100, 110, 120>>
```

Comment:
I think we can use list here and later use `Enum.map/2` and `Enum.each/2` to avoid repeats and numbers in the names.

Reply:
ok, i refactored this test